backend/pkg/logger/data/logger.go

```diff
@@ -0,0 +1,115 @@
+package data
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/logger"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/packet/data"
+)
+
+type ValueName string
+
+const (
+	name abstraction.LoggerName = "data"
+)
+
+type Logger struct {
+	running    bool
+	lock       sync.Mutex
+	valueFiles map[data.ValueName]*os.File
+}
+
+type DataRecord struct {
+	packet *data.Packet
+}
+
+func (data *DataRecord) Name() abstraction.LoggerName {
+	return name
+}
+
+func (sublogger *Logger) Start() error {
+	sublogger.lock.Lock()
+	defer sublogger.lock.Unlock()
+
+	if sublogger.running {
+		fmt.Printf("Logger already running")
+		return nil
+	}
+	sublogger.newFiles()
+
+	sublogger.running = true
+	fmt.Printf("Logger started")
+	return nil
+}
+
+type num interface {
```

Comment:
```suggestion
type numeric interface {
```
and make a comment explaining why we use this interface